### ISSUE TITLE
putty shims open a blank window

### ISSUE THREAD

**Issue by trajano** [lukesampson/scoop-extras#2801]
When running putty from the Win-R dialog it starts up the shim which in turn opens up a Window that does not close.

I tried simply removing the "putty.exe" from the "bin" list but that made it so that I can't start "putty.exe" by typing it on the command line.

I was wondering if instead of a "shim" it creates a link to the binary.  But I can't find an option to do that.

**Comment by rasa** [comment]
Can you test if the shim replacement fixes this issue: https://github.com/71/scoop-better-shimexe ?

**Comment by trajano** [comment]
@rasa how? (I can't find instructions on how to do this yet, I just took some educated guesses)

**Comment by rasa** [comment]
```batch
wget https://github.com/71/scoop-better-shimexe/releases/download/1.0/shim.exe
wget https://raw.githubusercontent.com/71/scoop-better-shimexe/master/repshims.bat
call repshims.bat
```

**Comment by trajano** [comment]
nope
![image](https://user-images.githubusercontent.com/110627/64199642-c3447c00-ce58-11e9-8a3e-0b676290c4dd.png)

**Comment by trajano** [comment]
Its actually worse in a way.  Closing the shim also closes putty.  Looking at the code, it does not do any check whether it is a windowed app that it is running or not.

However, I think it would be cleaner to just add symlinks instead of shims.  But I am not sure why that was not used, there may be a technical reason I don't know yet.

**Comment by 71** [comment]
> Its actually worse in a way. Closing the shim also closes putty. Looking at the code, it does not do any check whether it is a windowed app that it is running or not.
> 
> However, I think it would be cleaner to just add symlinks instead of shims. But I am not sure why that was not used, there may be a technical reason I don't know yet.

Shims can also pass some arguments to the `.exe` they redirect to, which may be required in some instances (I've never personally encountered it, but I know the `.shim` format allows it). AFAIK this is not possible with symlinks.

(We could use symlinks in most cases and shims in the few others, but I didn't design the shims so I might be missing details as to why this design was chosen instead).

**Comment by rasa** [comment]
> However, I think it would be cleaner to just add symlinks instead of shims. But I am not sure why that was not used, there may be a technical reason I don't know yet.

See https://github.com/lukesampson/scoop/issues/1606#issuecomment-392108771

Thanks for testing the C-based shim. I’m sorry it didn’t solve the issue.

**Comment by trajano** [comment]
Good news on from https://github.com/71/scoop-better-shimexe it handles this situation correctly now.  Perhaps we should switch the shims to use this better exe.

The only thing I would recommend to change is to use symlinks to the shim so we do not waste drive space for the duplicate shim-exes

**Comment by rasa** [comment]
This issue should be fixed in lukesampson/scoop#3998, which was merged into master. To use, type:
```
scoop config shim kiennq
scoop reset *
```
or
```
scoop config shim 71
scoop reset *
```
Eventually, one of those shims (probably kiennq) we be made the default. 

If this issue is still a problem after trying both solutions above, please reopen this issue. Thanks!
